circus-train-housekeeping/src/test/java/com/hotels/bdp/circustrain/housekeeping/LegacyReplicaPathRepositoryTest.java

```diff
@@ -0,0 +1,118 @@
+package com.hotels.bdp.circustrain.housekeeping; /**
+ * Copyright (C) 2016-2017 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.joda.time.Instant.parse;
+import static org.junit.Assert.assertThat;
+
+import java.io.File;
+import java.util.List;
+
+import org.joda.time.Instant;
+import org.junit.AfterClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.config.DelegatingApplicationContextInitializer;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.test.annotation.Rollback;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.support.DependencyInjectionTestExecutionListener;
+import org.springframework.test.context.support.DirtiesContextTestExecutionListener;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.github.springtestdbunit.DbUnitTestExecutionListener;
+import com.github.springtestdbunit.TransactionDbUnitTestExecutionListener;
+import com.github.springtestdbunit.annotation.DatabaseSetup;
+import com.github.springtestdbunit.annotation.DbUnitConfiguration;
+import com.github.springtestdbunit.annotation.ExpectedDatabase;
+import com.github.springtestdbunit.assertion.DatabaseAssertionMode;
+
+import com.hotels.bdp.circustrain.housekeeping.model.CircusTrainLegacyReplicaPath;
+import com.hotels.housekeeping.model.LegacyReplicaPath;
+import com.hotels.housekeeping.repository.LegacyReplicaPathRepository;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@TestPropertySource(properties = {
+    "spring.jpa.database=H2",
+    "spring.jpa.hibernate.ddl-auto=update",
+    "spring.jpa.hibernate.generate-ddl=true",
+    "spring.datasource.initialize=true",
+    "spring.datasource.schema=classpath:/schema.sql",
+    "spring.datasource.max-wait=10000",
+    "spring.datasource.max-active=5",
+    "spring.datasource.test-on-borrow=true",
+    "housekeeping.expired-path-duration=P3D",
+    "housekeeping.data-source.driver-class-name=org.h2.Driver",
+    "housekeeping.data-source.url=jdbc:h2:./target/data/housekeeping;AUTO_SERVER=TRUE;DB_CLOSE_ON_EXIT=FALSE",
```

Comment:
Is there no way this can be specified programatically so we could use a temp folder and get auto cleanup on it?

Reply:
There is cleanup on it if you look at the teardown() method in this class

Reply:
Yeah, but we've had to write that code and also hardcode a path under the target folder, if we were able to use the location from a JUnit temp folder we'd get all that "for free".